diesel/src/query_source/mod.rs

```diff
@@ -82,6 +82,17 @@ pub trait Table: QuerySource + AsQuery + Sized {
     fn all_columns() -> Self::AllColumns;
 }
 
+/// Determines how many times `Self` appears in `QS`
+///
+/// This trait is primarily used to determine whether or not a column is
+/// selectable from a given from clause. A column can be selected if its table
+/// appears in the from clause *exactly once*.
+///
+/// We do not allow the same table to appear in a query multiple times in any
+/// context where referencing that table would be ambiguous (depending on the
+/// context and backend being used, this may or may not be something that would
+/// otherwise result in a runtime error).
```

Comment:
This is great! I'm pretty sure people will get error messages with `AppearsInFromClause` in them, and if the search for that, these are the docs they'll find. I can't think of any right now, but if we had a few "usual error cases" we could describe here (with fixes), that could be quite helpful. (Like the [error index](https://doc.rust-lang.org/1.21.0/error-index.html).)